pages/_app.js

```diff
@@ -2,69 +2,150 @@ import useSWR from "swr";
 import GlobalStyle from "../styles";
 import { useEffect, useState } from "react";
 import useLocalStorageState from "use-local-storage-state";
-
+import AlarmTimer from "@/components/AlarmTimer";
 const fetcher = (url) => fetch(url).then((res) => res.json());
 
 export default function App({ Component, pageProps }) {
+  const [settings, setSettings] = useLocalStorageState("settings", {
+    defaultValue: {
+      countryFlags: {
+        USD: true,
+        EUR: true,
+        GBP: true,
+        AUD: false,
+        CAD: false,
+        NZD: false,
+        CHF: false,
+        JPY: false,
+        CNY: false,
+      },
+      flagsTurnedOn: false,
+      alarmTriggerA: 5,
+      alarmTriggerB: 20,
+      eventDuration: 30,
+      impactLow: true,
+      impactMedium: true,
+      impactHigh: true,
+      bankHolidays: true,
+    },
+  });
+
   const [newsEvents, setNewsEvents] = useState([]);
   const [alarmEvents, setAlarmEvents] = useLocalStorageState("events", {
     defaultValue: [],
   });
   const { data: events } = useSWR("/api/fetchThisWeek", fetcher);
 
+  function changeSettings(setting, value) {
```

Comment:
I feel like we can simplify this function a little bit. Inside every if block (or else if) we are declaring a new variable `updatedSettings`. We could move this declaration to the top of the function. Also, inside every if (or else if) we call `setSettings`. We could move this function call to after the `if else`. 

i.e. something like this. 

``` js
  function changeSettings(setting, value) {
    const updatedSettings = { ...setting };
    if (setting === "flag") {
      updatedSettings.countryFlags[value] =
        !updatedSettings.countryFlags[value];
    } else if (setting === "preferredCurrenciesToggle") {
      updatedSettings.flagsTurnedOn = !updatedSettings.flagsTurnedOn;
    } else if (setting === "alarmTriggerA") {
      updatedSettings.alarmTriggerA = value;
    } else if (setting === "alarmTriggerB") {
      updatedSettings.alarmTriggerB = value;
    } else if (setting === "eventDuration") {
      updatedSettings.eventDuration = value;
    } else if (setting === "impactLow") {
      updatedSettings.impactLow = !updatedSettings.impactLow;
    } else if (setting === "impactMedium") {
      updatedSettings.impactMedium = !updatedSettings.impactMedium;
    } else if (setting === "impactHigh") {
      updatedSettings.impactHigh = !updatedSettings.impactHigh;
    } else if (setting === "bankHolidays") {
      updatedSettings.bankHolidays = !updatedSettings.bankHolidays;
    }
    setSettings(updatedSettings);
  }
```

Just an idea, and i haven't tested the code.